app/views/onboarding/Welcome.tsx

```diff
@@ -0,0 +1,247 @@
+import React, { useEffect, useState } from 'react';
+import {
+  TouchableOpacity,
+  Image,
+  ImageBackground,
+  StyleSheet,
+  View,
+} from 'react-native';
+import { useNavigation } from '@react-navigation/native';
+import { useTranslation } from 'react-i18next';
+
+import {
+  LOCALE_LIST,
+  getUserLocaleOverride,
+  setUserLocaleOverride,
+  supportedDeviceLanguageOrEnglish,
+} from './../../locales/languages';
+import { NativePicker } from '../../components/NativePicker';
+import { Typography } from '../../components/Typography';
+import EulaModal from './EulaModal';
+import { useStatusBarEffect } from '../../navigation';
+
+import { Images } from '../../assets';
+import {
+  Forms,
+  Buttons,
+  Spacing,
+  Colors,
+  Typography as TypographyStyles,
+} from '../../styles';
+
+type Agreement = 'Agreed' | 'NotAgreed';
+
+const Onboarding = (): JSX.Element => {
+  useStatusBarEffect('light-content');
+  const { t } = useTranslation();
+  const navigation = useNavigation();
+  const [locale, setLocale] = useState(supportedDeviceLanguageOrEnglish());
+  const [modalVisible, setModalVisible] = useState(false);
+  const [eulaAgreement, setEulaAgreement] = useState<Agreement>('NotAgreed');
+
+  useEffect(() => {
+    getUserLocaleOverride().then((locale) => {
+      if (locale) {
+        setLocale(locale);
+      }
+    });
+  });
+
+  const toggleCheckbox = () => {
```

Comment:
Is there a reason we are using strings instead of storing the agreement state in a boolean? Then we could do something like `setEulaAgreementAccepted(!eulaAgreementAccepted)`? Line 32 makes sense coming from Elm, but since there are only two states why not just true/false?

Reply:
totally, I go back n forth on this. We chatted irl, but for future readers: On the one hand, having true/false is significantly less code (potentially low signal to noise ratio code at that), on the other hand by providing a named type we better express the domain concept. `The user has agreed to the terms` is more clear than `the user has set the terms agreement state to true` and it also leads us open to future iterations where we might have more than just 2 states, for example, there might be a world were we want to force an active choice with something like: `type EulaAgreement = "UnAnswered" | "Agreed" | "Disagreed"` and having the union type now allows for this easily.